pkg/server/register.go

```diff
@@ -43,7 +43,7 @@ const defaultName = "m-${System Information/Manufacturer}-${System Information/P
 var (
 	sanitize   = regexp.MustCompile("[^0-9a-zA-Z]")
 	doubleDash = regexp.MustCompile("--+")
-	start      = regexp.MustCompile("^[a-zA-Z]")
+	start      = regexp.MustCompile("^[a-zA-Z0-9]")
```

Comment:
I think it was used just to check if the label (was the name) had a valid first character, otherwise `m` is prepended.
Yep, this code is a bit cumbersome maybe it deserves a rework 😅

Reply:
Then as a result of this change a name staring with a digit has no `m` prepended right? will it be an issue?

Reply:
AFAIK no, kubernetes resource names and labels may start with a digit too

Reply:
yep, this was done only with the machine name in mind, so the regex never expected to find a machine name starting with a digit :D

This is mostly due to protection against someone setting a machine name like `-sdfa` or an empty name "" 

And yeah k8s names/labels are alphanumeric+"-" and "_" but not at the start or end -> https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names